ouroboros-consensus-cardano/src/byron/Ouroboros/Consensus/Byron/Ledger/Mempool.hs

```diff
@@ -162,6 +163,12 @@ instance HasTxId (GenTx ByronBlock) where
   txId (ByronUpdateProposal i _) = ByronUpdateProposalId i
   txId (ByronUpdateVote     i _) = ByronUpdateVoteId     i
 
+instance ConvertRawTxId (GenTx ByronBlock) where
+  toRawTxIdHash (ByronTxId             i) = CC.abstractHashToShort i
+  toRawTxIdHash (ByronDlgId            i) = CC.abstractHashToShort i
+  toRawTxIdHash (ByronUpdateProposalId i) = CC.abstractHashToShort i
+  toRawTxIdHash (ByronUpdateVoteId     i) = CC.abstractHashToShort i
```

Comment:
This instance assumes that it is not possible for two different constructors to contain the same hash. This seems reasonable, as the payload of these different kinds of transactions are completely different. However, to be extra safe, we could eg prepend a tag byte.

Reply:
This seems fine, especially as Byron transactions aren't being newly submitted anymore.